### PR TITLE
Persistent-topic

### PR DESCRIPTION
**Pull request by piotrek6641**
Sequence that persistently stores given amount of data in a specified time.

### CODE REVIEW

File: js/persistency-sequence-generate-topic/index.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+                    res();
+                });
+            }
+
+            if (dataAmount > 0) {
+                yield new Uint8Array(dataAmount).fill(0);
+            }
+
+            console.log(`finished in: ${(Date.now() - startTime).toString()} ms`);
```
Comment (patuwwy) [inline js/persistency-sequence-generate-topic/index.ts:38]: Detailed summary :)

```suggestion
            console.log(`Finished in: ${(Date.now() - startTime).toString()} ms. Bytes generated: ${a} x ${dataSize} + ${dataAmount} = ${a * dataSize + dataAmount}`);
```

File: js/persistency-sequence-generate-topic/index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+const exp: (TransformApp | { provides: string; contentType: string })[] = [
+    async function(_input, executionTime: number = 1 * 10 * 1000, dataAmount: number = 16 * 1000) {
+        const out = new PassThrough();
+        const dataSize = 16;
+        const intervalDelay = executionTime / (dataAmount / dataSize);
+
+        async function* gen() {
+            let a = 0;
+
+            const startTime = Date.now();
```
Comment (patuwwy) [inline js/persistency-sequence-generate-topic/index.ts:15]:
```suggestion
            const startTime = Date.now();
            
            console.log(`Generating ${dataSize} bytes every ${intervalDelay}ms ...`);
```

File: js/persistency-sequence-generate-topic/index.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+            while (dataAmount > dataSize) {
```
Comment (patuwwy) [inline js/persistency-sequence-generate-topic/index.ts:17]:
```suggestion
            while (dataAmount >= dataSize) {
```

File: js/persistency-sequence-generate-topic/index.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            if (dataAmount > 0) {
+                yield new Uint8Array(dataAmount).fill(0);
+            }
+
+            console.log(`finished in: ${(Date.now() - startTime).toString()} ms`);
+        }
+
+        this.on("new-test-start", () => {
+            Readable.from(gen()).pipe(out);
+        });
```
Comment (patuwwy) [inline js/persistency-sequence-generate-topic/index.ts:43]: The output will end when the generator goes out. At that time, Instance will also be terminated. If you want to trigger this event multiple times you need to pipe Readable with the `{ end: false }` option. 

```suggestion
        this.on("new-test-start", () => {
            Readable.from(gen()).pipe(out, { end: false });
        });
```

File: js/persistency-sequence-generate-topic/package.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "persistecy-sequence-generate-topic",
```
Comment (patuwwy) [inline js/persistency-sequence-generate-topic/package.json:2]: not sure if it shouldn't be named "@scramjet/persistecy-sequence-generate-topic"

**Review by patuwwy — CHANGES_REQUESTED** [review]
see comments Bro

File: js/persistency-sequence-generate-topic/index.ts (hunk 4 above)
Comment (patuwwy) [reply js/persistency-sequence-generate-topic/index.ts:43]:
```suggestion
        this.on("new-test-start", () => {
            if (!busy) {
                Readable.from(gen()).pipe(out);
            } else {
                console.log("I'm busy, man...");
            }
        });
```

**Comment by piotrek6641** [comment]
> see comments Bro

Thank you for suggestions bro :)

**Review by patuwwy — APPROVED** [review]

**Pull request merged**